linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -683,10 +681,21 @@ private[optimizer] abstract class OptimizerCore(config: CommonPhaseConfig) {
 
     val thisType = if (arrow) NoType else AnyType
 
-    val (allNewParamsWithUse, newBody) = transformIsolatedBody(None, thisType,
-        captureParams ++ params ++ restParam, AnyType, body, scope.implsBeingInlined)
+    val (inlinableCaptures, normalCaptureParams) = {
+      val (i, n) = captureParams.iterator.zip(tcaptureValues.iterator).partition {
+        _._2 match {
+          // TODO: Add more criteria here
+          case PreTransLocalDef(LocalDef(_, _, _: ReplaceWithThis | _: ReplaceWithConstant)) => true
```

Comment:
@sjrd You said [here](https://github.com/scala-js/scala-js/issues/4653#issuecomment-1064739460) that `this` is just like any `val`. Does that mean we'd just include `ReplaceWithVarRef` and maybe `ReplaceWithRecordVarRef` here too?